File: examples/sdk/nodejs/1_fluent.ts
// Fluent builder: override provider, profile, or vault URL
import { Envilder, SecretProviderType } from '@envilder/sdk';

const secrets = await Envilder.fromMapFile('../../../secrets-map.json')
  .withProvider(SecretProviderType.Aws)
  .withProfile('mac')
  .resolve();

for (const [key, value] of secrets) {
  console.log(`${key} = ${value}`);
}

File: examples/sdk/python/1_fluent.py
# /// script
# dependencies = ["envilder>=0.4.0"]
# ///

# Fluent builder: override provider, profile, or vault URL
from envilder import Envilder, SecretProviderType

secrets = (
    Envilder.from_map_file("../../../envilder.json")
    .with_provider(SecretProviderType.AWS)
    .with_profile("mac")
    .resolve()
)

for key, value in secrets.items():
    print(f"{key} = {value}")
